myenum.rb

```diff
@@ -0,0 +1,263 @@
+
+    arr = to_a
+    i = 0
+    loop do
+      yield(arr.to_a[i])
+      i += 1
+      break if i == arr.to_a.length
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = 0
+    my_each do |item|
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    temparr = []
+
+    my_each do |temp|
+      temparr.push(temp) if yield(temp) == true
+    end
+    temparr
+  end
+
+  def my_all?(arg = nil)
+    return true if empty?
+
+    if block_given?
+      my_each { |temp| return false if yield(temp) == false }
+      return true
+    elsif arg.nil?
+      my_each { |temp| return false if temp.nil? || temp == false }
+    elsif !arg.nil? && arg.instance_of?(Regexp)
+      my_each { |temp| return false unless temp.match(arg) }
+    elsif !arg.nil? && (arg.is_a? Class)
+      my_each { |temp| return false unless [temp.class, temp.class.superclass].include?(arg) }
+    else
+      my_each { |temp| return false unless temp == arg }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    return false if empty?
+
+    if block_given?
+      my_each { |temp| return true if yield(temp) == true }
+      return false
+    elsif arg.nil?
+      my_each { |temp| return true if temp }
+    elsif !arg.nil? && arg.instance_of?(Regexp)
+      my_each { |temp| return true if temp.match(arg) }
+    elsif !arg.nil? && (arg.is_a? Class)
+      my_each { |temp| return true if [temp.class, temp.class.superclass].include?(arg) }
+    else
+      my_each { |temp| return true if temp == arg }
+    end
+    false
+  end
+
+  def my_none?(arg = nil)
+    return true if empty?
+
+    if block_given?
+      my_each { |temp| return false if yield(temp) == true }
+      return true
+    elsif arg.nil?
+      my_each { |temp| return false if temp }
+    elsif !arg.nil? && arg.instance_of?(Regexp)
+      my_each { |temp| return false if temp.match(arg) }
+    elsif !arg.nil? && (arg.is_a? Class)
+      my_each { |temp| return false if [temp.class, temp.class.superclass].include?(arg) }
+    else
+      my_each { |temp| return false if temp == arg }
+    end
+    true
+  end
+
+  def my_count(args = nil)
+    count = 0
+    if block_given?
+      my_each do |temp|
+        count += 1 if yield(temp) == true
+      end
+      count
+
+    elsif !args.nil?
+      my_each do |temp|
+        count += 1 if temp == args
+      end
+      count
+
+    else
+      count = length
+    end
+  end
+
+  def my_map(myproc = nil)
+    return to_enum(:my_map) unless block_given? || !myproc.nil?
+
+    i = 0
+    temparr = []
+    if myproc.nil?
+      my_each do |temp|
+        temparr[i] = yield(temp)
+        i += 1
+      end
+    else
+      my_each do |y|
+        temparr[i] = myproc.call(y)
+        i += 1
+      end
+    end
+    temparr
+  end
+
+  def my_inject(initial = nil, sym = nil, &prc)
+    return self if empty?
```

Comment:
line 128 is preventing the my_inject method for pursuing given blocks and it returns undefined method empty?